src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -462,15 +462,34 @@ private static boolean isNonLocalTypeDeclaration(DetailAST typeDeclAst) {
     private static DetailAST getBlockContainingLocalAnonInnerClass(DetailAST literalNewAst) {
         DetailAST currentAst = literalNewAst;
         DetailAST result = null;
-        while (!TokenUtil.isOfType(currentAst, CONTAINERS_FOR_ANON_INNERS)) {
-            if (currentAst.getType() == TokenTypes.LAMBDA
-                    && currentAst.getParent()
-                    .getParent().getParent().getType() == TokenTypes.OBJBLOCK) {
+        while (result == null) {
+            if (!isTokenNestedUnderContainerForAnonInners(currentAst)) {
                 result = currentAst;
-                break;
             }
             currentAst = currentAst.getParent();
-            result = currentAst;
+        }
+        return result;
+    }
+
+    /**
+     * Whether the ast is nested under tokens of type
+     * {@link UnusedLocalVariableCheck#CONTAINERS_FOR_ANON_INNERS} or {@link TokenTypes#LAMBDA}.
```

Comment:
Please remove mentions of specific tokens, such a list is bound to get outdated/ignored, and is an implementation detail that the caller should not have to worry about.